surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -302,17 +273,24 @@ pcl::MarchingCubes<PointNT>::performReconstruction (pcl::PointCloud<PointNT> &po
 
   // Run the actual marching cubes algorithm, store it into a point cloud,
   // and copy the point cloud + connectivity into output
-  points.clear ();
+  intermediate_cloud.clear();
+
+  // preallocate memory assuming a hull. suppose 6 point per voxel
+  intermediate_cloud.reserve((size_t)(res_y_*res_z_ + res_x_*res_z_ + res_x_*res_y_) * 2 * 6);
```

Comment:
I wanted to avoid massive copy of vector. How to check overflow? Could you show me a file with this function? (in voxel grid I have seen similar output, but for overflow of uint32 or int32)